integration/yarn/src/main/java/tachyon/yarn/ApplicationMaster.java

```diff
@@ -296,21 +353,27 @@ private void launchTachyonWorkerContainers(List<Container> containers) {
         FormatUtils.getSizeFromBytes((long) mRamdiskMemInMB * Constants.MB));
 
     for (Container container : containers) {
-      if (mAllWorkersAllocatedLatch.getCount() == 0) {
-        break;
-      }
-      try {
-        // Launch container by create ContainerLaunchContext
-        ContainerLaunchContext ctx = Records.newRecord(ContainerLaunchContext.class);
-        ctx.setCommands(commands);
-        ctx.setEnvironment(environmentMap);
-        LOG.info("Launching container {} for Tachyon worker {} on {} with worker command: {}",
-            container.getId(), mNumWorkers - mAllWorkersAllocatedLatch.getCount(),
-            container.getNodeHttpAddress(), command);
-        mNMClient.startContainer(container, ctx);
-        mAllWorkersAllocatedLatch.countDown();
-      } catch (Exception ex) {
-        LOG.error("Error launching container " + container.getId() + " " + ex);
+      synchronized (mWorkerHosts) {
+        if (mWorkerHosts.size() >= mNumWorkers
```

Comment:
It should never be possible - we only make enough requests to reach `mNumWorkers` workers, then wait for them all to complete before making another round of requests. I'll remove the condition.